src/test/java/GmailServletTest.java

```diff
@@ -91,32 +182,394 @@ public void setUp() throws IOException, GeneralSecurityException {
         .thenReturn(AUTHENTICATION_VERIFIED);
 
     request = Mockito.mock(HttpServletRequest.class);
-    response = Mockito.mock(HttpServletResponse.class);
+    response = new HttpServletResponseFake();
     Mockito.when(request.getCookies()).thenReturn(validCookies);
+  }
+
+  /**
+   * Auxiliary method to get a Message payload (with a "From" header) given a sender's email. From
+   * header in the form of: "From": "<email@email.com>"
+   *
+   * @param email the sender's email
+   * @return a MessagePart instance that can be used as the payload of a Message
+   */
+  private static MessagePart generateMessagePayload(String email) {
+    return new MessagePart()
+        .setHeaders(
+            Collections.singletonList(
+                new MessagePartHeader().setName("From").setValue(String.format("<%s>", email))));
+  }
+
+  /**
+   * Auxiliary method to get a Message payload (with a "From" header) given a sender's email. From
+   * header in the form of: "From": "SenderName <email@email.com>"
+   *
+   * @param email the sender's email
+   * @param contactName the name of the sender
+   * @return a MessagePart instance that can be used as the payload of a Message
+   */
+  private static MessagePart generateMessagePayload(String email, String contactName) {
+    return new MessagePart()
+        .setHeaders(
+            Collections.singletonList(
+                new MessagePartHeader()
+                    .setName("From")
+                    .setValue(String.format("%s <%s>", contactName, email))));
+  }
+
+  /**
+   * Auxiliary method to get a GmailResponse object from a servlet when using doGet
+   *
+   * @param request Mock HttpRequest
+   * @param response Mock HttpResponse
+   * @return GmailResponse object from doGet method
+   * @throws IOException if a read/write issue occurs while processing the request
+   * @throws ServletException if another unexpected issue occurs while processing the request
+   */
+  private GmailResponse getGmailResponse(
+      HttpServletRequest request, HttpServletResponseFake response)
+      throws IOException, ServletException {
+    servlet.doGet(request, response);
+    return gson.fromJson(response.getStringWriter().toString(), GmailResponse.class);
+  }
+
+  @Test
+  public void noNDaysParameter() throws IOException, ServletException {
```

Comment:
I was reading our internal doc on Java practices and it says it is okay to simply throw `Exception` from tests

Reply:
Just updated - sorry for missing this comment!